src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs

```diff
@@ -134,12 +136,27 @@ protected static async Task<string> RenderRazorPageAsync(Compilation compilation
 
         // Create ViewContext.
         var appBuilder = WebApplication.CreateBuilder();
-        appBuilder.Services.AddMvc().AddApplicationPart(assembly);
+        appBuilder.Services.AddMvc().ConfigureApplicationPartManager(manager =>
+        {
+            var partFactory = new ConsolidatedAssemblyApplicationPartFactory();
+            foreach (var applicationPart in partFactory.GetApplicationParts(assembly))
+            {
+                manager.ApplicationParts.Add(applicationPart);
+            }
+        });
```

Comment:
Part of test infra changes from https://github.com/dotnet/razor/pull/8520 to make test rendering work here.